drivers/modem/modem_cmd_handler.c

```diff
@@ -672,3 +672,21 @@ int modem_cmd_handler_init(struct modem_cmd_handler *handler,
 
 	return 0;
 }
+
+void modem_cmd_handler_disable_eol(struct modem_cmd_handler *handler)
+{
+	struct modem_cmd_handler_data *data;
+	data = (struct modem_cmd_handler_data *)(handler->cmd_handler_data);
```

Comment:
I think this should be more threadsafe: the gsm_ppp function call this directly.

Reply:
Adding the modem cmd handler TX lock to the function results in the recover function no longer working due to strange implementations of the GSM_PPP driver.